scripts/mods/msu/persistent_data_manager.nut

```diff
@@ -0,0 +1,99 @@
+this.persistent_data_manager <- {
+	m = {
+		ModConfigPath = "mod_config/"
+		Mods = {}
+	},
+	
+	function create()
+	{
+		this.importModConfigFiles();
+	}
+
+	function importModConfigFiles()
+	{
+		local persistentDirectory = this.IO.enumerateFiles(this.m.ModConfigPath);
+		if (persistentDirectory == null)
+		{
+			return;
+		}
+		foreach (file in persistentDirectory)
+		{
+			local components = split(file, "/");
+			local modID = components[1];
+			local settingsType = components[2];
+			::MSU.Mod.Debug.printWarning(format("Checking file, potential modID: '%s' and settingstype '%s'.", modID, settingsType), "persistence");
+			this.setMod(modID);
+			if (!(settingsType in this.getMods()[modID]))
+			{
+				this.getMods()[modID][settingsType] <- file;
+			}
+		}
+	}
+	
+
+	function getMods()
+	{
+		return this.m.Mods;
+	}
+
+	function hasMod( _modID )
+	{
+		return (_modID in this.getMods());
+	}
+
+	function getMod(_modID)
+	{
+		if (!this.hasMod(_modID))
+		{
+			throw ("Mod " + _modID + " not found in mods!");
```

Comment:
should probably be keyNotFound or maybe a logError followed by a keyNotFound

Reply:
Yes KeyNotFound, with my suggested function with arg system :D